src/commands/build/wranglerjs/mod.rs

```diff
@@ -39,26 +47,82 @@ pub fn run_build(project: &Project) -> Result<(), failure::Error> {
         let wranglerjs_output: WranglerjsOutput =
             serde_json::from_str(&output).expect("could not parse wranglerjs output");
 
-        if wranglerjs_output.has_errors() {
-            message::user_error(wranglerjs_output.get_errors().as_str());
-            failure::bail!("Webpack returned an error");
-        }
+        write_wranglerjs_output(&bundle, &wranglerjs_output)
+    } else {
+        failure::bail!("failed to execute `{:?}`: exited with {}", command, status)
+    }
+}
 
-        bundle
-            .write(&wranglerjs_output)
-            .expect("could not write bundle to disk");
+pub fn run_build_and_watch(
+    project: &Project,
+    tx: Option<Sender<()>>,
+) -> Result<(), failure::Error> {
+    let (mut command, temp_file, bundle) = setup_build(project)?;
+    command.arg("--watch=1");
 
-        let msg = format!(
-            "Built successfully, built project size is {}",
-            wranglerjs_output.project_size()
-        );
+    info!("Running {:?} in watch mode", command);
 
-        fs::remove_file(temp_file)?;
-        message::success(&msg);
-        Ok(())
-    } else {
-        failure::bail!("failed to execute `{:?}`: exited with {}", command, status)
+    thread::spawn(move || {
+        let _command_guard = util::GuardedCommand::spawn(command);
+
+        let (watcher_tx, watcher_rx) = channel();
+        let mut watcher = watcher(watcher_tx, Duration::from_secs(1)).unwrap();
+
+        watcher.watch(&temp_file, RecursiveMode::Recursive).unwrap();
```

Comment:
I guess you don't need Recursive mode here since it's just a file

Reply:
it just gets ignored, but I will change it since it makes the intent unclear! thanks for the catch